src/index.d.ts

```diff
@@ -0,0 +1,256 @@
+
+export function highByte(value: number): number;
+
+export function bitRead(value: number, bitnum: number): number;
+
+export function bitWrite(value: number, bitnum: number, bitdata: any): number;
+
+export function bitSet(value: number, bitnum: number): number;
+
+export function bitClear(value: number, bitnum: number): number;
+
+export function bit(bitnum: number): number;
+
+export function min(x: number, y: number): number;
+
+export function max(x: number, y: number): number;
+
+export function abs(x: number): number;
+
+export function constrain(x: number, a: number, b: number): number;
+
+export function map(value: number, fromLow: number, fromHigh: number, toLow: number, toHigh: number): number;
+
+export function pow(x: number, y: number): number;
+
+export function sqrt(x: number): number;
+
+export function sin(radians: number): number;
+
+export function cos(radians: number): number;
+
+export function tan(radians: number): number;
+
+export function randomSeed(x: number): number;
+
+export function random(min: number, max?: number): number;
+
+export function delay(...args: any[]): any;
+
+export function autorun(...args: any[]): any;
+
+export function serverStart(...args: any[]): any;
+
+export function socketJSReqHandler(...args: any[]): any;
+
+export function addSocketListeners(...args: any[]): any;
+
+export function setGlobals(...args: any[]): any;
+
+// CONSTANTS
+export const OUTPUT: 'out';
+export const INPUT: 'in';
+export const INPUT_PULLUP: 'in_pullup';
+export const ANALOG_OUTPUT: 'analog_out';
+export const HIGH: 1;
+export const LOW: 0;
+export const LSBFIRST: 1;
+export const MSBFIRST: 0;
+export const CHANGE: 'both';
+export const RISING: 'rising';
+export const FALLING: 'falling';
+
+// INTERFACES - adjusted based on observation / testing
+export interface PinModeObj {
+    gpio?: {
+        active: boolean,
+        direction: 'in' | 'out',
+        allocated: boolean,
+    };
+    modes: string[];
+    mux?: number;
+    name: string;
+    pin: string;
+    pinState: string;
+    pullup?: 'diabled' | 'pullup' | 'pulldown';
+    pwm?: {
+        freq: number,
+        value: number,
+    };
+    rx?: 'enabled' | 'disabled';
+    slew?: 'fast' | 'slow';
+}
+
+export interface BoneObj {
+    i2c: i2cObj;
+    uarts: UartsObj;
+    getPinKeys(): string[];
+    getPinObjects(): any;
+    naturalCompare(): any;
+}
+
+export interface PinsObj {
+    [i: string]: PinInfo;
+}
+
+export interface UartsObj {
+    [i: string]: UartInfo;
+}
+
+export interface i2cObj {
+    [i: string]: i2cInfo;
+}
+
+export interface PinInfo {
+    // Analog In
+    ain?: number;
+    // GPIO / LEDs / Analog In
+    eeprom?: number;
+    // GPIO / LEDs
+    gpio?: number;
+    key: string;
+    // LEDs
+    led?: string;
+    mux?: string;
+    muxRegOffset?: string;
+    // All
+    name: string;
+    options?: string[];
+    // PWM
+    pwm?: PwmInfo;
+    scale?: number;
+    universalName?: string;
+}
+
+export interface PwmInfo {
```

Comment:
Really don't think this should be a public interface. I should probably hide it.